### PR TITLE
Research tweaks - easier botany, icons, etc.

### PR DESCRIPTION
**Pull request by ThePiachu**
## About The Pull Request

I've wanted to make Research and especially the hydroponics side of things a bit more user friendly for a while. So I tweaked a few things.

To start, the various Research papers have different icons now:

image

Experiments have a "?", meaning they only pose a question. Synthesis looks like a bubbling solution . Analysis has a "!!", meaning its information is complete. Research Grants have small green stamps on them to indicate money. Hopefully these will help Researchers figure out what's on their floor.

Then, I've made the plant stuff a bit easier:
- Genes tell you what category they encode (since CM isn't a botany-focused game and Research doesn't have an hour to spend on just messing with plant genes)
- You can rename seed packets with a verb. This will also rename any plants you grow from those seeds, so if you have some kind of Uranium-producing Wheat, you can call it Uranium Wheat and you will know exactly what's going on.

These two changes should make Research plant growing much more accessible for newcomers.

## Why It's Good For The Game

Research currently has problems with being a bit obtuse, so lowering the complexity in the systems that are way too complicated for CM definitely helps ease new players into them. Visual clarity on the icons will help sorting through the research papers a bit more - you'll no longer have a mixed floor of papers and not know what is what.

## Changelog

:cl: ThePiachu, Triiodine
imageadd: Made the various Research-related papers a bit more visually distinct from one another.
qol: Research grants now delete themselves after they are cashed in.
qol: Hydro trays now display their basic information when examined.
qol: You can now rename mutated seeds by right-clicking on their packets. Should come in very handy in Research!
balance: Plant genes now tell you what they relate to for easier gene splicing (CM isn't a botany-focused game, so we cut our Researchers some slack there!).
/:cl:

### CODE REVIEW

File: code/modules/paperwork/paper.dm [hunk 1]
```diff
@@ -764,17 +768,22 @@
 			info += "<BR>Overdoses at: [S.overdose] units\n"
 			info += "<BR>Standard duration multiplier of [REAGENTS_METABOLISM/S.custom_metabolism]x</font><BR>\n"
 			completed = TRUE
+			icon_state = "paper_wy_full_report"
 		else
 			info += "CLASSIFIED:<I> Clearance level [S.gen_tier] required to read the database entry.</I><BR>\n"
+			icon_state = "paper_wy_partial_report"
 	else if(S.chemclass == CHEM_CLASS_SPECIAL && !chemical_data.clearance_x_access && !info_only)
 		info += "CLASSIFIED:<I> Clearance level <B>X</B> required to read the database entry.</I><BR>\n"
+			icon_state = "paper_wy_partial_report"
```
Comment (harryob) [inline code/modules/paperwork/paper.dm:777]:
```suggestion
		icon_state = "paper_wy_partial_report"
```
quick one for the CI fail

**Review by fira — APPROVED** [review]

File: code/game/objects/items/reagent_containers/food/snacks/grown.dm [hunk 2]
```diff
@@ -31,6 +31,7 @@
 /obj/item/reagent_container/food/snacks/grown/proc/update_from_seed()// Fill the object up with the appropriate reagents.
 	if(!isnull(plantname))
 		var/datum/seed/S = seed_types[plantname]
+		name = S.seed_name //Copies the name from the seed, important for renamed plants
```
Comment (Geevies) [inline code/game/objects/items/reagent_containers/food/snacks/grown.dm:34]: put this after the check to see if S exists, you can check if chems exists after that

File: code/modules/hydroponics/hydro_tray.dm [hunk 3]
```diff
@@ -643,6 +643,44 @@
 		anchored = !anchored
 		to_chat(user, "You [anchored ? "wrench" : "unwrench"] \the [src].")
 
+/obj/structure/machinery/portable_atmospherics/hydroponics/get_examine_text(mob/user)
+	. = ..()
+	var/hyrdro_info = show_hydro_info(user)
+	if(hyrdro_info)
+		. += hyrdro_info
+
+
```
Comment (Geevies) [inline code/modules/hydroponics/hydro_tray.dm:652]:
```suggestion
```

File: code/modules/hydroponics/hydro_tray.dm [hunk 4]
```diff
@@ -643,6 +643,44 @@
 		anchored = !anchored
 		to_chat(user, "You [anchored ? "wrench" : "unwrench"] \the [src].")
 
+/obj/structure/machinery/portable_atmospherics/hydroponics/get_examine_text(mob/user)
+	. = ..()
+	var/hyrdro_info = show_hydro_info(user)
```
Comment (Geevies) [inline code/modules/hydroponics/hydro_tray.dm:648]: hyrdro

File: code/modules/hydroponics/hydro_tray.dm [hunk 5]
```diff
@@ -643,6 +643,44 @@
...
+	if(hyrdro_info)
+		. += hyrdro_info
+
+
+/obj/structure/machinery/portable_atmospherics/hydroponics/proc/show_hydro_info(mob/user as mob)
+	var/info = ""
+	if(seed && !dead)
+		info += "[src] has \blue [seed.display_name] \black planted.\n"
```
Comment (Geevies) [inline code/modules/hydroponics/hydro_tray.dm:656]: use spans instead of inbuilt color modifiers

File: code/modules/hydroponics/hydro_tray.dm [hunk 6]
```diff
@@ -643,6 +643,44 @@
...
+	else
+		info += "[src] is empty.\n"
+	info += "Water: [round(waterlevel,0.1)]/100\n"
+	info += "Nutrient: [round(nutrilevel,0.1)]/10\n"
+	if(weedlevel >= 5)
+		info += "[src] is \red filled with weeds!\n"
+	if(pestlevel >= 5)
+		info += "[src] is \red filled with tiny worms!\n"
+
+	if(!istype(src,/obj/structure/machinery/portable_atmospherics/hydroponics/soil))
```
Comment (Geevies) [inline code/modules/hydroponics/hydro_tray.dm:668]: i know you're just moving this code, but please make this not do istype(src), just make a proc override or something

File: code/modules/hydroponics/seeds.dm [hunk 7]
```diff
@@ -20,6 +20,33 @@
...
 	return ..()
 
+/obj/item/seeds/verb/rename_seeds() //set amount_per_transfer_from_this
+	set name = "Rename Seeds"
+	set category = "Object"
+	set src in usr
+	if(!ishuman(usr))
+		return
+	var/mob/living/carbon/human/user = usr
+	var/obj/item/seeds/S = user.get_active_hand()
```
Comment (Geevies) [inline code/modules/hydroponics/seeds.dm:30]: not sure about forcing it to be in their active hand, either hand it to be on their person or adjacent them

File: code/modules/hydroponics/seeds.dm [hunk 8]
```diff
@@ -20,6 +20,33 @@
...
+	if(!istype(S))
+		return
```
Comment (Geevies) [inline code/modules/hydroponics/seeds.dm:32]: send the user a error message here so it doesn't just fail silently

File: code/modules/hydroponics/seeds.dm [hunk 9]
```diff
@@ -20,6 +20,33 @@
...
+	if (seed.roundstart)
+		to_chat(usr, SPAN_WARNING("You can't rename basic seeds!"))
```
Comment (Geevies) [inline code/modules/hydroponics/seeds.dm:34]:
```suggestion
		to_chat(user, SPAN_WARNING("You can't rename basic seeds!"))
```

File: code/modules/hydroponics/seeds.dm [hunk 10]
```diff
@@ -20,6 +20,33 @@
...
+		return
+	var/new_name = copytext(reject_bad_text(input(user,"Rename seed variety?", "Set new seed variety name", "")), 1, MAX_NAME_LEN)
```
Comment (Geevies) [inline code/modules/hydroponics/seeds.dm:36]: tgui input here, probably

File: code/modules/hydroponics/seeds.dm [hunk 11]
```diff
@@ -20,6 +20,33 @@
...
+		return
+	S.seed.seed_name = new_name
+	//Sure, the various seed types have different names for the plant thing but honestly it's too much work changing that for what it's worth...
+	//So generic name instead!
+	S.seed.display_name = "[new_name] plant"
+	for (var/obj/item/seeds/SS in GLOB.seed_list)
+		if (SS.seed.uid == S.seed.uid)
+			SS.update_appearance() //updates the name
+
+	to_chat(usr, SPAN_INFO("Variety #[seed.uid] renamed to \"[new_name]\""))
```
Comment (Geevies) [inline code/modules/hydroponics/seeds.dm:47]:
```suggestion
	to_chat(user, SPAN_INFO("Variety #[seed.uid] renamed to \"[new_name]\""))
```

Comment (Geevies) [inline code/modules/hydroponics/seeds.dm:47]: also, message this to admins so if someone uses a naughty word, admins can hop on it

**Review by Geevies — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/seeds.dm (hunk 10 above)
Comment (ThePiachu) [reply code/modules/hydroponics/seeds.dm:37]: I tried searching for TGUI text input, but I don't think it exists. There is a numerical one and a list one, but not string.

I checked character preferences for how that UI handles name changes and it's

`var/raw_name = input(user, "Choose your character's name:", "Character Preference")  as text|null`

so I don't think there is another way of handling it other than `input`?

Comment (ThePiachu) [reply code/modules/hydroponics/seeds.dm:37]: From Discord:
"harryob — Today at 12:00 AM
oh uhhh i still need to port that"

Comment (ThePiachu) [reply code/modules/hydroponics/seeds.dm]: Apparently tgui text input was added recently, so fixed it as requested.

File: code/modules/hydroponics/hydro_tray.dm [hunk 12]
```diff
@@ -643,6 +643,43 @@
...
+	var/hydro_info = show_hydro_info(user)
+	if(hydro_info)
+		. += hydro_info
+
+/obj/structure/machinery/portable_atmospherics/hydroponics/proc/show_hydro_info(mob/user as mob)
+	var/info = ""
+	if(seed && !dead)
+		info += "[src] has " + SPAN_HELPFUL("[seed.display_name]") +" planted.\n"
+		if(plant_health <= (seed.endurance / 2))
+			info += "The plant looks \red unhealthy.\n"
```
Comment (harryob) [inline code/modules/hydroponics/hydro_tray.dm:657]:
```suggestion
			info += "The plant looks " + SPAN_RED("unhealthy") + ".\n"
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Swendvideo** [comment]
Oh my god YES! YEEEEEEES!

**Comment by ThePiachu** [comment]
Updated the icons as per feedback:
image

**Comment by ThePiachu** [comment]
Proper icons by Triiodine:

image

**Review by nauticall — APPROVED** [review]
that's what im talkin about! sprites look good now, here's my 👍 of approval

**Review by harryob — APPROVED** [review]

**Pull request merged**